src/components/CityBlock/CityBlock.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from "react";
+import { Button, Card, message, Switch } from "antd";
+import { PlusOutlined } from "@ant-design/icons";
+
+import { CityBlockContent } from "./CityBlockContent";
+
+import { getWeatherByCity } from "../../service/weather/weather.service";
+
+import style from "./city-block.module.scss";
+
+export const CityBlock = ({
+  cityName,
+  handleAddCityModalOpen,
+  handleRemoveCity,
+  isPinned,
+  handlePinCity,
+}) => {
+  const [currWeather, setCurrWeather] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    if (cityName) {
+      getData();
+    } else {
+      setCurrWeather(null);
+    }
+  }, [cityName]);
+
+  const getData = async () => {
+    setIsLoading(true);
+    try {
+      const weatherResponse = await getWeatherByCity(cityName);
+      if (weatherResponse.ok) {
+        const weather = await weatherResponse.json();
+        setCurrWeather(weather);
+      } else {
+        handleRemoveCity();
```

Comment:
it is better to ask for reload instead of removing city, but this requirement wasn't specified, so this implementation is also OK :)